semantic_release/vcs_helpers.py

```diff
@@ -19,7 +20,8 @@
 try:
     repo = Repo(".", search_parent_directories=True)
 except InvalidGitRepositoryError:
-    repo = None
+    # repo = None
+    raise GitError("Not in a valid git repository")
```

Comment:
Why do we need to raise an error here?

Reply:
does it make sense to run python-semantic-release outside of a valid git repository ?
I agree it could be managed diffrently. 
Generally I prefer an error message followed by exit rather noop.

Reply:
This code runs on import as a part of a command and thus buy passes the error handling.

Reply:
@relekang mypy is failing because of your initial implement
 
https://github.com/relekang/python-semantic-release/runs/4270162688?check_suite_focus=true#step:5:8

Maybe you need to change error handling implement. Or not initialize repo variable in global scope ...

Reply:
instead of assigning None a simple pass will do, because it will set repo to None implicitly and mypy won't complain. And it's safe to do that, because the `check_repo' wrapper checks for repo for None and raises an exception.
Instead of using a global variable, it would be better to just wrap the whole thing in a class.